router/usecase/routertesting/suite.go

```diff
@@ -344,6 +346,12 @@ func (s *RouterTestHelper) SetupRouterAndPoolsUsecase(mainnetState MockMainnetSt
 
 	tokensUsecase.RegisterPricingStrategy(domain.ChainPricingSourceType, pricingSource)
 
+	// Set up Coingecko pricing strategy
+	options.PricingConfig.DefaultSource = domain.CoinGeckoPricingSourceType
+	coingeckoPricingSource, err := pricing.NewPricingStrategy(options.PricingConfig, tokensUsecase, routerUsecase)
```

Comment:
Should we instead be registering a mock coingecko pricing strategy?

Otherwise, our unit tests may start making network requests which is not ideal, leading to flakiness and increased latency